src/common/Utils.ts

```diff
@@ -7,94 +7,108 @@
 import * as vscode from "vscode";
 import { EXTENSION_ID } from "../client/constants";
 
-export function getSelectedSnippet(): string {
-  const editor = vscode.window.activeTextEditor;
-  if (!editor) {
-    return "";
-  }
-  const selection = editor.selection;
-  const text = editor.document.getText(selection);
-  return text;
+export function getSelectedCode(editor: vscode.TextEditor): string {
+    if (!editor) {
+        return "";
+    }
+    const selection = editor.selection;
+    const text = editor.document.getText(selection);
+    return text;
 }
 
+export function getSelectedCodeLineRange(editor: vscode.TextEditor): { start: number, end: number } {
+    if (!editor) {
```

Comment:
What you will return if the user selects empty space without any code?

Reply:
If the selected code length is non-zero, it will return the value for that line irrespective of content.

Reply:
Update: It now handles the empty selection and label will not be shown in such scenarios.